pkg/workceptor/workceptor.go

```diff
@@ -480,57 +495,79 @@ func (w *Workceptor) GetResults(unitID string, startPos int64, doneChan chan str
 
 			break
 		}
-		var stdout *os.File
-		var err error
 		filePos := startPos
+		statChan := make(chan struct{}, 1)
```

Comment:
No need to buffer and send empty structs and sent empty structs into it. Just close the channel at the sender and the receiver will unblock